### PR TITLE
maintainers: fix IDs and update github handles

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

These were found with a `nix eval ... | jq ... | xe -s 'gh api ... && echo'` oneliner which bash failed to preserve in my history. I then got a merge conflict with #259990, where `SamirTalwar` erroneously was renamed. There was a typo in his githubId.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by eclairevoyant — APPROVED** [review]
Thanks for catching! Will have to see how the script can be improved for future use.

Only question I have is, when should names be updated? I see in many cases, people use a pseudonym on gh, but another name on the maintainers list, and am not sure when such mismatches should be left alone or updated (eg picnoir in this list).

Anyway, none of these changes appear incorrect after manually checking myself, so I won't block the review on this.

**Comment by pbsds** [comment]
I think the principle of least merge conflict is a good stance. Maintainers are however welcome to themselves treewide change their `maintainers.<attribute>`.

I decided not to drop any maintainer in this PR, even though there were candidates. I think we should discuss whether it is acceptable to drop removed accounts from `maintainer-list.nix` or not at all. Git is not the only info they provide.

**Comment by eclairevoyant** [comment]
Github is the only place to contribute, and maintainership requires keeping aware of and interacting PRs and issues related to their package. (Which requires a GH account.)

So if their account is deleted, it does not make sense for them to stay as package maintainers, it is just bad data quality.

**Comment by pbsds** [comment]
I updated both the github handle and the name of Ocean, because the old name struck me as a dead name. (Only the name change is visible post-merge however.) For picnoir whether this is the case or not was not as apparent to me, it could be a refresh of the online identity alone and not the IRL one. I decided not to dig.

Touching `maintainer-list.nix` in general can quickly get finicky, and clear guidelines should be discussed and put in place.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -5973,7 +5973,7 @@
   fortuneteller2k = {
     email = "lythe1107@gmail.com";
     matrix = "@fortuneteller2k:matrix.org";
-    github = "fortuneteller2k";
+    github = "moni-dz";
     githubId = 20619776;
     name = "fortuneteller2k";
   };
```
Comment (Artturin) [inline maintainers/maintainer-list.nix:5979]: https://github.com/NixOS/nixpkgs/pull/268077

Comment (eclairevoyant) [reply maintainers/maintainer-list.nix:5979]: name should be `"moni"` as well

**Pull request merged**